iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -659,26 +660,23 @@ popo::InterfacePortData* PortManager::acquireInterfacePortData(capro::Interfaces
     }
 }
 
-void PortManager::publishCurrentServiceRegistry() const noexcept
+void PortManager::publishServiceRegistry() const noexcept
 {
-    // Send the new serviceRegistry here and ring the bell to inform all ServiceDiscovery instances about the change
     if (m_serviceRegistryPublisherPortData.has_value())
     {
         PublisherPortUserType publisher(m_serviceRegistryPublisherPortData.value());
-        auto maybeChunkHeader = publisher.tryAllocateChunk(sizeof(ServiceRegistry),
-                                                           alignof(ServiceRegistry),
-                                                           CHUNK_NO_USER_HEADER_SIZE,
-                                                           CHUNK_NO_USER_HEADER_ALIGNMENT);
-
-        if (!maybeChunkHeader.has_error())
-        {
-            auto sample = static_cast<ServiceRegistry*>(maybeChunkHeader.value()->userPayload());
-
-            // Copy the complete registry
-            *sample = m_serviceRegistry;
-
-            publisher.sendChunk(maybeChunkHeader.value());
-        }
+        publisher
+            .tryAllocateChunk(sizeof(ServiceRegistry),
+                              alignof(ServiceRegistry),
+                              CHUNK_NO_USER_HEADER_SIZE,
+                              CHUNK_NO_USER_HEADER_ALIGNMENT)
+            .and_then([&](auto& chunk) {
+                auto sample = static_cast<ServiceRegistry*>(chunk->userPayload());
+
+                *sample = m_serviceRegistry;
```

Comment:
Here a comment would be in order that this copy is not concurrent to any modification of `m_serviceRegistry` (as it happens in the same thread where those modifications are happening) and hence safe.

The problem is this does not hold anymore if anyone modifies it elsewhere in the future. Not much we can do about this to enforce it though.

Reply:
Added